### ISSUE TITLE
[HOLD for payment 2021-12-13] [HOLD for payment 2021-12-06] After deleting the last message, editing the next last message doesn't update the message displayed in LHN

### ISSUE THREAD

**Issue by dklymenk** [Expensify/App#3743]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Send 2 messages (message 1 and message 2) in a chat
2. Delete the last message (message 2)
3. Edit the next last message (message 1)

## Expected Result:
The LHN should be updated after editing last message regardless of your previous actions

## Actual Result:
The LHN is not updated

## Workaround:
Sending a new message and editing it updates LHN properly

## Platform:
Where is this issue occurring?

 Web :heavy_check_mark: 
 iOS :heavy_check_mark: 
 Android :heavy_check_mark: 
 Desktop App :heavy_check_mark: 
 Mobile Web :heavy_check_mark: 
 
 ## Proposal:
The issue is caused by unreliable values in `reportMaxSequenceNumbers` in `src/libs/actions/Report.js`. This array is not updated when you delete a message. 
 
 Here is how I would rewrite `updateReportActionMessage` to update the `reportMaxSequenceNumbers`.
 
 ```js
 function updateReportActionMessage(reportID, sequenceNumber, message) {
    const actionToMerge = {};
    actionToMerge[sequenceNumber] = {message: [message]};
    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionToMerge);

    // If this is the most recent message
    if (sequenceNumber === reportMaxSequenceNumbers[reportID]) {
        // If message is deleted we need to update maxSequenceNumber
        if (message.html === '') {
            const connectionID = Onyx.connect({
                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
                callback: (reportActionList) => {
                    // To make sure the callback is only called once
                    Onyx.disconnect(connectionID);

                    // Find last non-empty message
                    const lastReportAction = !_.isEmpty(reportActionList)
                        ? _.find(Object.values(reportActionList).reverse(),
                            reportAction => _.last(reportAction.message).html !== ''
                            && reportAction.created !== 'CREATED')
                        : null;

                    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
                        maxSequenceNumber: lastReportAction.sequenceNumber,
                    });
                },
            });
        } else {
            // Update the lastMessageText in the report object
            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
                lastMessageText: message.html,
            });
        }
    }
}
```

I would also need to check how other functions across the app interact with `maxSequenceNumber` to make sure reducing it doesn't break anything.

**Version Number:** 1.0.68-4
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @strepanier03 (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by NikkiWines** [comment]
A note that this issue is related to the changes made in this PR: https://github.com/Expensify/Expensify.cash/pull/3526

**Comment by dklymenk** [comment]
Is there any way to push this one forward? It's blocking PR #3526 and all the code is already there, I'm just waiting for a green light to go ahead with the implementation.

**Comment by parasharrajat** [comment]
So this is related to changes in #3526 which NikkiWines mentioned above. So I think you are good to go here to submit the PR for this. As this part of old PR.

**Comment by parasharrajat** [comment]
But if you are waiting for your solution to be reviewed then waiting is good.

**Comment by NikkiWines** [comment]
Hmm, it looks like @strepanier03 is out-of-office currently. Going to add the `External` label so this issue can be triaged appropriately

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mallenexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by mallenexpensify** [comment]
Everything _looks_ good to me, I'm going to assign an engineer to review.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @iwiznia (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mallenexpensify** [comment]
Job posted https://www.upwork.com/jobs/~01a6f9be9af612f0cd
@iwiznia please review the proposal then assign to to @dklymenk

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by mallenexpensify** [comment]
@puneetlath assigned to you via the `External` label because I'm out next week

**Comment by iwiznia** [comment]
That proposal seems it would work, but I am a bit concerned about the one time connect call to get all actions of the report of the action being updated.
Do you think we could instead put the logic to update the maxSequenceNumber [here](https://github.com/Expensify/Expensify.cash/blob/135ab15684f71a88c29ed085b3ca6e1d997c88ca/src/libs/actions/ReportActions.js#L34) instead where we should already have all the data available without having to do that one time connect call @dklymenk?

**Comment by NikkiWines** [comment]
Adding the Help Wanted label, as it looks like we're still accepting new proposals on this one (please remove the label if it's not applicable)

**Comment by dklymenk** [comment]
@iwiznia 

> Do you think we could instead put the logic to update the maxSequenceNumber here instead where we should already have all the data available without having to do that one time connect call @dklymenk?

There are a few issues at play here.

1. Since all this Onyx stuff is async, we'll never get up-to-date information in context of one function, where an `Onyx.merge` call updates something in local storage. The values that are updated via `Onyx.connect` subscribers outside that function will only be updated after the function finished execution. 

Let's say we move my suggested code (including this `Onyx.merge` call below) to [the place you suggested](https://github.com/Expensify/Expensify.cash/blob/135ab15684f71a88c29ed085b3ca6e1d997c88ca/src/libs/actions/ReportActions.js#L34) 
```js
 Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
     maxSequenceNumber: lastReportAction.sequenceNumber,
 });
 ```
               
The value we care about in [updateReportActionMessage](https://github.com/Expensify/Expensify.cash/blob/main/src/libs/actions/Report.js#L499-L510) is `reportMaxSequenceNumbers[reportID]` that is updated by an `Onyx.connect` subscriber [here](https://github.com/Expensify/Expensify.cash/blob/main/src/libs/actions/Report.js#L1191-L1207). Due to the nature of event-driven architecture and async functions, after we call `Onyx.merge` to update the report action, neither [this](https://github.com/Expensify/Expensify.cash/blob/135ab15684f71a88c29ed085b3ca6e1d997c88ca/src/libs/actions/ReportActions.js#L22-L36) callback gets called nor does [this one](https://github.com/Expensify/Expensify.cash/blob/main/src/libs/actions/Report.js#L1191-L1207) until `updateReportActionMessage` function finishes it's execution, so the value `reportMaxSequenceNumbers[reportID]` will still be unreliable.

This is why regardless of where exactly we put that code, in the end we will still need to do a `Onyx.connect` inside the function to get up-to-date information about report actions or report.

2. The main reason for me to do it like this is because I'm not just fetching all report actions to get max sequence number there, but also to potentially get last message text/html in order to update LHN. This suggestion for me is not just a fix for the issue described in the OP, but also an infrastructure development for this PR #3526. Since I'll need to fetch all reportActions via `Onyx.connect` to get new last message details after last message deletion anyway, I see no harm in doing it that way. If my proposal in this issue gets approved, the code from #3526 will fit in flawlessly, like a puzzle piece.

Thanks.

**Comment by tgolen** [comment]
I'd like to put a little bit of a hold on this. It seems to me that this is complex code, and both the code being proposed here and in https://github.com/Expensify/Expensify.cash/pull/3526 are only making it more and more complex.

I think it would be worth it to take a step back, look at some of the root causes of these problems, and see if there is a better solution.

Remember, this stuff was originally developed without the understanding that messages would be deleted. That's OK, we avoid pre-optimizing solutions and over-engineering things. This means that when new functionality needs to be added (like deleting report actions), we refactor the original solutions to allow the new functionality to exist.

It seems like some of the root problems that lead up to this are:
1. There is data being duplicated in Onyx. Specifically, the last message details exist in both the report collection and the report action collection. This leads to complex code that must now keep both of those in sync
2. When the last message is deleted, the report's `maxSequenceNumber` isn't updated

Before talking any more about solutions, can we agree that these are the root cause of the problem? Am I missing anything?

**Comment by dklymenk** [comment]
Yes, I 100% agree with your concerns. While my proposed code fixes the user facing issues, the code is unreasonably complex for a feature as simple last message display in LHN.

> 1. There is data being duplicated in Onyx. Specifically, the last message details exist in both the report collection and the report action collection. This leads to complex code that must now keep both of those in sync

If I was to redesign this flow, I would remove the last message data from report collection and extract last non-empty message from report action collection directly in component that needs it. This will allow us to simplify functions `updateReportActionMessage` and `getSimplifiedReportObject` in `lib/actions/Report.js` as well eliminate the need for code proposed for `deleteReportComment` in #3526 .

> 2. When the last message is deleted, the report's maxSequenceNumber isn't updated

The vague proposal above will also make it so we don't need to care about `maxSequenceNumber`, at least in the context of LHN. There will still be some code that uses it, but that issue should IMO be addressed separately, since it won't interfere with LHN in any way.

Please let me know, if that is something you are looking for. I'll happily draft a proposal with actual code.

Thanks.

**Comment by tgolen** [comment]
Awesome, thanks for confirming my thoughts!

> If I was to redesign this flow, I would remove the last message data from report collection and extract last non-empty message from report action collection directly in component that needs it.

This is what was running through my head as well. I think we should definitely consider this. Could you work on providing a proposal to do this? I _think_ only the LHN is impacted by this, but we need to be careful that nothing else is.

> The vague proposal above will also make it so we don't need to care about maxSequenceNumber, at least in the context of LHN. There will still be some code that uses it, but that issue should IMO be addressed separately, since it won't interfere with LHN in any way.

Cool. I think if it removes the need to worry about `maxSequenceNumber` in this context, then I agree it should be addressed separately.

**Comment by marcaaron** [comment]
Please keep in mind that the "last message" is saved in the report object when the app first inits so we can populate the LHN in an efficient way. Sets of report actions notoriously:

1. Take time to fetch completely at large numbers
2. Are **not guaranteed** to even exist due to cache eviction

Separately, I'm currently working on a backend optimization that will _significantly_ improve the time required to fetch this information when the app inits and "refreshes". Which, again, largely exists so we can avoid having to fetch all the report actions for all the reports.

So, I'm not sure if I agree yet with the premise that report actions should be the single source of truth. But I agree there is some opportunity to unwind complexity here.

**Comment by tgolen** [comment]
Thanks for that insight, Marc!

**Comment by dklymenk** [comment]
Hello, again. Here is what I've tried:

1. Add `reportActions` to `withOnyx` in `SidebarLinks.js`.
2. Pass `this.props.reportActions` to `getSidebarOptions` function.
3. From `getSidebarOptions` pass `reportActions` to `getOptions`.
4. In `getOptions` right [here](https://github.com/Expensify/Expensify.cash/blob/main/src/libs/OptionsListUtils.js#L310) I've added the following code (it is basically a code from one of my PRs for getting last non-empty message details):
```js
        const currentReportActions = lodashGet(
            reportActions, `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
        );

        // Find last non-empty message
        const lastReportAction = !_.isEmpty(currentReportActions)
            ? _.find(Object.values(currentReportActions).reverse(),
                ({message, actionName}) => message.length && _.last(message).html !== ''
                && actionName !== 'CREATED')
            : null;

        // Populate last message details
        const lastMessageDetails = {};
        if (lastReportAction) {
            // Replace line breaks with spaces and remove all HTML from the message
            lastMessageDetails.lastMessageText = lastReportAction.isAttachment
                ? '[Attachment]'
                : Str.stripHTML(_.last(lastReportAction.message).html.replace(/((<br[^>]*>)+)/gi, ' '));
            lastMessageDetails.lastMessageTimestamp = lastReportAction.timestamp;
            lastMessageDetails.lastActorEmail = lodashGet(lastReportAction, 'actorEmail', '');
        } else {
            lastMessageDetails.lastMessageText = '';
            lastMessageDetails.lastMessageTimestamp = 0;
            lastMessageDetails.lastActorEmail = '';
        }
```
5. I have then replaced all the references to `report.lastMessageText`, `report.lastMessageTimestamp` and `report.lastActorEmail` with values from the new `lastMessageDetails` object in `getOptions` and in `createOption` functions.

I got to say, at the first glance it works and feels great. Here is a teaser:

https://user-images.githubusercontent.com/64093836/125136044-b1962480-e112-11eb-90f4-f8357351a9ba.mp4

It works flawlessly as long as you have reports actions for your reports in local storage. This is where a warning voiced by @marcaaron comes in. In my testing, if you relogin, the report actions are fetched inconsistently, sometimes I get them right away, sometimes it takes about 5-10 seconds and sometimes only the most recent one (aka default route) is loaded. Here is example video of this behavior (notice how `api?command=Report_GetHistory` requests for non-default chat are fetched only at the end of the video):

https://user-images.githubusercontent.com/64093836/125137012-86143980-e114-11eb-9562-60a90a9030cf.mp4

To me, if I sort out the report actions loading, I should be able to get an all around consistent behavior, but in light of what @marcaaron said, it might be impossible or there are some other issues with this approach I haven't yet discovered.

So, my question is: Should I continue digging in this direction? Maybe the "backend optimization" @marcaaron mentions allows for faster report actions fetching and local storage population? Or should we abandon this refactoring and continue with my original proposal?

**Comment by marcaaron** [comment]
>Maybe the "backend optimization" @marcaaron mentions allows for faster report actions fetching and local storage population

No, it will not touch the flows that populate `reportActions` in local storage at all.

>it might be impossible or there are some other issues with this approach I haven't yet discovered

There are. Without a refactor of how things work now + consideration for overall performance the suggestion to use `reportActions` in storage to populate the LHN will have unexpected consequences.

After thinking about this for a bit I think maybe everything here is working as expected though and there might be a misconception that **when a comment is deleted we should show the previous one**. 

Presently, when we delete a comment its value is set to an empty string which is why the "last message" appears to be incorrect or shows a user email instead of message. We are setting the text of the message to be an empty string here...

https://github.com/Expensify/Expensify.cash/blob/9d77e53f89537820c6e1af177c58a6aac55c4da4/src/libs/actions/Report.js#L1078-L1092

It seems like the original intention was for these comments to display in the UI as `[Comment deleted]` when they have replies attached to them. So should we use a similar strategy here? Otherwise, another solution could be to work backwards from the end of the list to find a comment that is not "deleted" (but that probably will require passing the next acceptable comment when we are deleting the last action and via `ReportActionsView` - which can access the full list of actions).

**Comment by dklymenk** [comment]
> After thinking about this for a bit I think maybe everything here is working as expected though and there might be a misconception that when a comment is deleted we should show the previous one.

The reason I was fixed on this idea is that the "expected result" section on the original [issue](https://github.com/Expensify/Expensify.cash/issues/3293) says "LHN should show the message that was before the deleted message.". If this is not set in stone, it becomes an infinitely easier problem to tackle. Changing these 2 lines is enough to get this behavior going (I would need to make tests to make sure IOU report actions are not marked as deleted comment or anything):
```diff
--- a/src/libs/actions/Report.js
+++ b/src/libs/actions/Report.js
@@ -166,7 +166,7 @@ function getSimplifiedReportObject(report) {
     // We convert the line-breaks in html to space ' ' before striping the tags
     const lastMessageText = lodashGet(lastReportAction, ['message', 'html'], '')
         .replace(/((<br[^>]*>)+)/gi, ' ')
-        .replace(/(<([^>]+)>)/gi, '');
+        .replace(/(<([^>]+)>)/gi, '') || '[Comment deleted]';
     const reportName = lodashGet(report, ['reportNameValuePairs', 'type']) === 'chat'
         ? getChatReportName(report, chatType)
         : report.reportName;
@@ -504,7 +504,7 @@ function updateReportActionMessage(reportID, sequenceNumber, message) {
     // If this is the most recent message, update the lastMessageText in the report object as well
     if (sequenceNumber === reportMaxSequenceNumbers[reportID]) {
         Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-            lastMessageText: message.html,
+            lastMessageText: message.html || '[Comment deleted]',
         });
     }
 }
```

> Otherwise, another solution could be to work backwards from the end of the list to find a comment that is not "deleted" (but that probably will require passing the next acceptable comment when we are deleting the last action and via ReportActionsView - which can access the full list of actions).

That's actually what my proposed code in #3293 is doing. Only difference is that instead of passing `reportActions` from `ReportActionsView` to `ReportActionItem` to `deleteReportComment`, it gets `reportActions` from local storage directly in `deleteReportComment`. There a few issues with this approach, the most important one being the fact that `deleteReportComment` is only run when user deletes his message, it doesn't get executed when other chat participants delete their messages. This is why if we are to do it like that, we **must** do it in `updateReportActionMessage` function that is called by pusher both when you edit/delete a message or your chat buddy does. This means we can't pass `reportActions` from component and we forced to do `Onyx.connect` to get report actions.

-----------------------
So, the way I see it, there are 2 solutions now:
1. My original proposal in the first post on this issue + migrating the code suggested in #3293 to `updateReportActionMessage`, to make it so both edits and deletions are handled by that function.
2. Change approach to showing `[Comment deleted]` instead of next last comment.

**Comment by iwiznia** [comment]
> The reason I was fixed on this idea is that the "expected result" section on the original issue says "LHN should show the message that was before the deleted message.". If this is not set in stone, it becomes an infinitely easier problem to tackle.

Yep, I think this is not set in stone and indeed seeing `message deleted` or similar might be better (and easier). So let's go with proposal 2 from your post above.

**Comment by dklymenk** [comment]
Great, I'll submit the PR shortly after getting hired on upwork.

It seems like the old job link is no longer available - https://www.upwork.com/jobs/~01a6f9be9af612f0cd.

**Comment by iwiznia** [comment]
I think it's due to our upgrade in upwork plan, right @mallenexpensify? Can you make sure to create one and hire @dklymenk please?

**Comment by dklymenk** [comment]
@iwiznia  I have created a PR #3979 since I have an active upwork contract for another issue the PR fixes. I assume there will be still another contract for this issue, right?

I have also made the `comment deleted` message translatable. Please check the spanish translation.

**Comment by iwiznia** [comment]
Yeah, once @mallenexpensify comes online, he can find or create the contract for you.

**Comment by iwiznia** [comment]
I think I found a bug in dev. A new chat with just an IOU split shows `[Comment deleted]` in the LHN and I think would happen the same if the last action is an IOU action:
![image](https://user-images.githubusercontent.com/521248/125324001-adc5f600-e2fc-11eb-90c0-585d09df8125.png)

@dklymenk can you please fix that?

**Comment by dklymenk** [comment]
Hmm, before sumbitting pull request I did test sending IOUs and attachments (aka empty messages), and didn't notice such an issue, I will take a look at sending IOU in a new chat in a couple of hours.

Thanks.

**Comment by dklymenk** [comment]
@iwiznia Hello. Right after you send a money request, the LHN is updated correctly. It is only after you refresh the page, you see a "[Comment deleted]" message. Basically, the root cause is not related to my PR and there is already an issue for it - #3737. Please take a look at my little explanation there if you need details on why this is happening. I am certain that the fix you guys are working on internally for #3737, will fix the behavior with "[Comment deleted]" after refresh too, given that it makes it so the html returned from BE is consistently non-empty for IOU report actions across all the endpoints.

It is very unfortunate that this is happening since "[Comment deleted]" is way more confusing for the end user than an email, so I would assume this means you're not deploying this PR until #3737 is resolved. Since that issue is internal now, there isn't much I can do to help sadly. Please let me know if I can be of help.

Thanks.

**Comment by mallenexpensify** [comment]
Created a new Upwork job https://www.upwork.com/jobs/~01bbcb01bf658bed62 @dklymenk , I invited you.  Let me know once you've applied and I'll hire.

**Comment by dklymenk** [comment]
Thanks. I've applied.

**Comment by mallenexpensify** [comment]
Hired @dklymenk in Upwork, assigned to this issue

**Comment by iwiznia** [comment]
Oh you are correct, thanks for the context!

**Comment by dklymenk** [comment]
UPDATE: #3979 has been reverted by #4085 because it wasn't ready to be merged until #3737 is fixed. Otherwise it creates an issue like #4073 which is way more destructive for UX than #3737.

So I've created a new PR #4133 that is meant to be merged after #3737 is fixed. After that this issue can be considered closed.

Thanks.

**Comment by iwiznia** [comment]
@dklymenk this is still on hold till #3737 is fixed?

**Comment by dklymenk** [comment]
@iwiznia  Unless the reasoning behind putting it hold hasn't changed, it is still on hold, yes.

If you merge it, you'll still see `[Comment deleted]` after sending an IOU and refreshing the page.

**Comment by mallenexpensify** [comment]
Added Hold to title with #3737 included.  Added Planning label and bumped to Weekly

**Comment by MelvinBot** [comment]
This issue has not been updated in over ***15*** days. @iwiznia, @mallenexpensify, @dklymenk eroding to ***Monthly*** issue.

P.S. Is everyone reading this sure this is really a near-term priority? Be brave: if you disagree, go ahead and close it out. If someone disagrees, they'll reopen it, and if they don't: one less thing to do!

**Comment by mallenexpensify** [comment]
still on hold

**Comment by mallenexpensify** [comment]
Getting closer... https://github.com/Expensify/App/issues/3737#issuecomment-913728569

**Comment by mallenexpensify** [comment]
So close https://github.com/Expensify/App/issues/3737#issuecomment-918991976
but... we're on hold for N6 for a week-ish so we likely have a lil while longer to wait.

**Comment by mallenexpensify** [comment]
Based off #3737 being closed ([PR for it](https://github.com/Expensify/Web-Expensify/pull/31586)), I think we can take this off hold now, right @iwiznia

**Comment by dklymenk** [comment]
Thanks for bringing it up. Apparently that fix is already deployed too.

I did quick check and my PR https://github.com/Expensify/App/pull/4133 still works. I'm marking it as ready for review now.

**Comment by dklymenk** [comment]
Yep, only thing holding the PR for this issue from being merged is the HOLD label on the issue itself. It can 100% be removed now as the fix for #3737 has been merged and deployed and the issue itself is closed.

**Comment by mallenexpensify** [comment]
I took the issue off hold in the title, we're still on `n6-hold` though so it might take a little bit of time for the PR to get merged, thanks for the patience @dklymenk (and... we'll add a bonus for the hold... once the PR is deployed to production and doesn't have regressions for 7 days)

**Comment by kadiealexander** [comment]
Please refer to [this post](https://expensify.slack.com/archives/C01GTK53T8Q/p1633557259458600) for updated information on the `n6 hold`, we've raised the bonus to **$250** for all issues where a **PR is created** before the N6 hold is lifted.

**Comment by mallenexpensify** [comment]
`n6-hold` is lifted, label removed!
@dklymenk can you provide an update on when you'll have a PR ready for review? Thanks

**Comment by dklymenk** [comment]
It's here

https://github.com/Expensify/App/pull/4133

**Comment by mallenexpensify** [comment]
Sorry for the delay @dklymenk , I commented on the PR and removed `[HOLD]` from the title so we should be good to go. 
https://github.com/Expensify/App/pull/4133#issuecomment-962219262

**Comment by botify** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.1.16-10 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/6458
- https://github.com/Expensify/App/pull/6458


If no regressions arise, payment will be issued on 2021-12-06. :confetti_ball:

**Comment by botify** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.1.17-7 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/6458
- https://github.com/Expensify/App/pull/6458


If no regressions arise, payment will be issued on 2021-12-13. :confetti_ball:

**Comment by mallenexpensify** [comment]
Paid @dklymenk in Upwork. Thanks for the help!
- $250 for job
- $250 for reporting
- $250 for Company Offsite Hold bonue.
